### PR TITLE
Support stopOnEntry and launchInTerminal in launch.json

### PR DESCRIPTION
**Pull request by testforstephen**
1. Add stopOnEntry in launch.json
2. Add console for lauchInTerminal feature in launch.json

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 - `projectName` - The preferred project in which the debugger searches for classes. There could be duplicated class names in different projects. This setting also works when the debugger looks for the specified main class when launching a program.
 - `cwd` - The working directory of the program.
 - `env` - The extra environment variables for the program.
+- `stopOnEntry` - Automatically stop program after launch.
+- `console` - Where to launch the program. Defaults to `internalConsole`.
```
Comment (yaohaizh) [inline README.md]: Is this term console provide by VSCode?  Normally, in this case, it will call  as `shell`?

File: package.json [hunk 2]
```diff
@@ -111,6 +111,26 @@
...
+                "type": "string",
+                "enum": [
+                  "internalConsole",
+									"integratedTerminal",
+									"externalTerminal"
+                ],
+                "enumDescriptions": [
+                  "VS Code Debug Console (which doesn't support to read input from a program)",
+                  "VS Code's integrated terminal",
+                  "external terminal that can be configured via user settings"
```
Comment (yaohaizh) [inline package.json]: `E`xternal

File: package.json [hunk 3]
```diff
@@ -111,6 +111,26 @@
...
+              },
+              "console": {
+                "type": "string",
+                "enum": [
+                  "internalConsole",
+									"integratedTerminal",
+									"externalTerminal"
+                ],
+                "enumDescriptions": [
+                  "VS Code Debug Console (which doesn't support to read input from a program)",
```
Comment (yaohaizh) [inline package.json]: support doing

File: README.md (hunk 1 above)
Comment (testforstephen) [reply README.md]: reference to vscode-node-debug implemented by vscode team, they used the term `console`.

Comment (akaroml) [reply README.md]: The specified console to launch the program. Defaults to ...

File: README.md [hunk 4]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+  - `internalConsole` - VS Code Debug Console (which doesn't support reading input from a program)
+  - `integratedTerminal` - VS Code's integrated terminal
```
Comment (akaroml) [inline README.md]: VS Code integrated terminal.

File: README.md [hunk 5]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 - `projectName` - The preferred project in which the debugger searches for classes. There could be duplicated class names in different projects. This setting also works when the debugger looks for the specified main class when launching a program.
 - `cwd` - The working directory of the program.
 - `env` - The extra environment variables for the program.
+- `stopOnEntry` - Automatically stop program after launch.
+- `console` - Where to launch the program. Defaults to `internalConsole`.
+  - `internalConsole` - VS Code Debug Console (which doesn't support reading input from a program)
```
Comment (akaroml) [inline README.md]: VS Code debug console (input stream not supported).

File: README.md [hunk 6]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+  - `integratedTerminal` - VS Code's integrated terminal
+  - `externalTerminal` - External terminal that can be configured via user settings
```
Comment (akaroml) [inline README.md]: via -> in

File: package.json [hunk 7]
```diff
@@ -111,6 +111,26 @@
                 "type": "object",
                 "description": "The extra environment variables for the program.",
                 "default": {}
+              },
+              "stopOnEntry": {
+                "type": "boolean",
+                "description": "Automatically stop program after launch.",
```
Comment (akaroml) [inline package.json]: Automatically pause the program after launching.

File: package.json [hunk 8]
```diff
@@ -111,6 +111,26 @@
...
+                "default": true
+              },
+              "console": {
+                "type": "string",
+                "enum": [
+                  "internalConsole",
+									"integratedTerminal",
```
Comment (akaroml) [inline package.json]: Mind the indention.

File: package.json [hunk 9]
```diff
@@ -111,6 +111,26 @@
...
+									"externalTerminal"
+                ],
+                "enumDescriptions": [
+                  "VS Code Debug Console (which doesn't support reading input from a program)",
```
Comment (akaroml) [inline package.json]: Please change accordingly.

**Pull request merged**